boards/m5stack/m5stack_cores3/m5stack_cores3_procpu.dts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2024 Zhang Xingtao <zhxt@live.cn>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include <espressif/esp32s3/esp32s3.dtsi>
+#include "m5stack_cores3-pinctrl.dtsi"
+
+/ {
+	model = "M5Stack CoreS3 PROCPU";
+	compatible = "m5stack,cores3";
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,console = &usb_serial;
+		zephyr,shell-uart = &usb_serial;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &slot0_partition;
+		zephyr,bt-hci = &esp32_bt_hci;
+	};
+
+	aliases {
+		i2c-0 = &i2c0;
+		watchdog0 = &wdt0;
+	};
+};
+
+&usb_serial {
+	status = "okay";
+};
+
+&uart0 {
+	status = "okay";
+	current-speed = <115200>;
+	pinctrl-0 = <&uart0_default>;
+	pinctrl-names = "default";
+};
+
+&i2c0 {
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+	pinctrl-0 = <&i2c0_default>;
+	pinctrl-names = "default";
+};
```

Comment:
Why not adding the BMI and BMM sensors?

Reply:
> Why not adding the BMI and BMM sensors?

Haven't get time on that two sensors, will check them later.

Reply:
@kartben 
I want to change the file structure later to support CoreS3 SE variant.
It's convinient for me to leave it as it is if you don't mind leaving that addressed in later.
What do you think?

Reply:
@zhxt 
Please create an issue with these sensors and aw9523b as a reminder.

Reply:
> Why not adding the BMI and BMM sensors?

BMI and BMM are now added, and CONFIG_I2C=y is  also added back.  @soburi @kartben